babeljs/jscontext.py

```diff
@@ -0,0 +1,102 @@
+try:
+    from py_mini_racer import py_mini_racer
+except ImportError:
+    py_mini_racer = None
+
+try:
+    from shutil import which
+except ImportError:
+    from distutils.spawn import find_executable as which
+
+import subprocess
+import json
+import tempfile
+import random
+import os
+
+
+if py_mini_racer:
+    ContextException = py_mini_racer.MiniRacerBaseException
+else:
+    class ContextException(Exception):
+        pass
+
+
+class NodeException(ContextException):
+    pass
+
+
+def get_context(use_mini_racer=None):
+    if use_mini_racer is None:
+        # Assume node if py_mini_racer is not installed
+        use_mini_racer = (py_mini_racer is not None)
+
+    if use_mini_racer:
+        if py_mini_racer is None:
+            raise ContextException("PyMiniRacer requested but not installed!")
+        return py_mini_racer.MiniRacer()
+    else:
+        if not which('node'):
+            raise NodeException("Node.js requested but not installed!")
+        return MinimalNodeWrapper()
+
+
+class MinimalNodeWrapper(object):
+    """
+    Bare-minimum node wrapper providing an API like MiniRacer.
+    """
+
+    def __init__(self):
+        self.code = ""
+
+    def eval(self, code):
+        # Note: Proper eval would actually execute code and return last result,
+        # but we don't need that in this case.
+        self.code += code
+
+    def call(self, function, *args):
+        # Dump "eval" code to file, run node, and parse result. Makes some
+        # assumptions about code structure, which are safe in this case.
+        js_fd, js_filename = tempfile.mkstemp('.js')
+        result_fd, result_filename = tempfile.mkstemp('.json')
```

Comment:
I think this will be a bit more intuitive and security:

```python
with tempfile.NamedTemporaryFile(suffix='.js') as js_fp:
    with tempfile.NamedTemporaryFile(suffix='.json') as res_fp:
        js_filename = js_fp.name
        res_filename = res_fp.name
        # balabala...

```

Reply:
Sure, I will see if I can make it work this way.  I think my reason for doing it the other way was to make sure the subprocess has access to write to res_filename.  (It's more of an issue on Windows).